### PR TITLE
Fix missing AuthIntent component causing queue backup

### PR DESCRIPTION
**Pull request by ShivamMistry**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
When a queued ACL assignment request can't find a AuthIntent component, the loop can be exited early causing it to backup indefinitely - this fixes that by continuing the loop instead of returning.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ void USpatialLoadBalanceEnforcer::ProcessQueuedAclAssignmentRequests()
 			// TODO(zoning): Not sure whether this should be possible or not. Remove if we don't see the warning again.
 			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("(%s) Entity without AuthIntent component will not be processed. EntityId: %lld"), *WorkerId, Request.EntityId);
 			CompletedRequests.Add(Request.EntityId);
-			return;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp]: Can you give any more details about when you're seeing this occur?

Comment (ShivamMistry) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp]: Seems to happen when moving over boundaries...
```
LogSpatialActorChannel: Delete entity request on 1008. Has authority: 0
LogSpatialActorChannel: Delete entity request on 1008. Has authority: 0
LogSpatialActorChannel: Delete entity request on 9008. Has authority: 0
LogSpatialActorChannel: Delete entity request on 9008. Has authority: 0
LogSpatialActorChannel: Delete entity request on 1009. Has authority: 0
LogSpatialActorChannel: Delete entity request on 1009. Has authority: 0
LogSpatialLoadBalanceEnforcer: Warning: (UnrealWorker3BA423B24905C3B9E6176AB5ABE3B852) Entity without AuthIntent component will not be processed. EntityId: 1008
LogSpatialLoadBalanceEnforcer: Warning: (UnrealWorker3BA423B24905C3B9E6176AB5ABE3B852) Entity without AuthIntent component will not be processed. EntityId: 9008
LogSpatialLoadBalanceEnforcer: Warning: (UnrealWorker3BA423B24905C3B9E6176AB5ABE3B852) Entity without AuthIntent component will not be processed. EntityId: 1009
```

Comment (cmsmithio) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp]: This looks good.  Also looks like this warning is normal if the entity is being deleted.  Maybe we should additionally change it to a log

**Comment by m-samiec** [comment]
This was fixed in another PR - thanks for the contribution 
https://github.com/spatialos/UnrealGDK/pull/1588

**Pull request closed without merging**